aws/scripts/py/cache_invalidation.py

```
@@ -0,0 +1,302 @@
+            domain for domain in origin_domains 
+            if "staging" in domain.lower() and "app." not in domain
+        ]
+        return len(staging_origins) > 0
+
+    def _classify_distribution(self, distribution: Dict[str, Any]) -> Optional[Environment]:
+        """Determine if distribution is staging, production, or should be skipped"""
+        dist_id = distribution.get('Id', 'Unknown')
+        
+        # Skip disabled distributions
+        if not distribution.get("Enabled", False):
+            self.logger.info(f"Skipping disabled distribution: {dist_id}")
```

Ruff (0.11.9)

Logging statement uses f-string

(G004)